### DISCUSSION TITLE
Support for web

### DISCUSSION THREAD

**Discussion by Arkantos900** [happy-turtle/oit-unity#45] [Q&A]
First of all, thanks for this amazing project, it works like a charm!

I need it to work for either WebGL or WebGPU so badly. I know it is specified that it doesn't work for those targets but I would like to know why and see if there is something I can do. If it is just more work, it would be great to know the next steps and maybe me and other devs can research and try.

Also, if it is basically impossible because how the project is coded is not compatible with web. then what other alternatives would you suggest?

Thank you very much!

**Comment by happy-turtle** [comment]
Hi, thanks for your kind words. Happy to hear the package works for you.

There is a specific reason why the algorithm does not run on WebGL: A Per-Pixel linked list requires `randomwrite` and compute shader capabilities for data structures like `RWStructuredBuffer`. WebGL 2.0 simply does not allow these features, it is equivalent to OpenGL ES 2, I think. You can have a look at [Unity's documentation to see that Shader Model 4](https://docs.unity3d.com/2022.3/Documentation/Manual/SL-ShaderCompileTargets.html) is required. That is just not compatible with WebGL 2.
With WebGPU these features are definitely possible though. 😉 Unity does not have a stable WebGPU backend yet. So you would have to try the experimental one: https://discussions.unity.com/t/early-access-to-the-new-webgpu-backend/933493. Let me know how far you get with this. Would be super cool to know if the project works on WebGPU. I can assist if you have any further questions. 🙂

**Comment by Arkantos900** [comment]
Thanks for the quick reply! 
Actually, one the first things that I tried was to build using WebGPU and it didn't work. At least not out-of-the-box, I just cloned this repo and built following the steps in that web gpu thread. 
Maybe it requires a couple of tweaks more, but at least I know that WebGPU is the path to follow. I'll keep trying and post if get any progress!
Thank you!!

**Comment by happy-turtle** [comment]
I've had some time to look at the currently available resources on WebGPU from Unity. It sounds like URP should be a good starting point, and to use Unity 6000.1 or newer. The manual states that [RWStructuredBuffers are indeed supported](https://docs.unity3d.com/6000.1/Documentation/Manual/WebGPU-limitations.html). It does not mention RWByteAddressBuffer though, not sure if that could be problematic. Compute shader are supported as well. Hearing all this, I would be fairly confident that by now, with the experimental WebGPU backend, the package should work. I will see if I get some time to try it out myself.
